test/lib/set-script.js

```diff
@@ -0,0 +1,154 @@
+const test = require('tap')
+const requireInject = require('require-inject')
+const setScriptDefault = require('../../lib/set-script.js')
+const parseJSON = require('json-parse-even-better-errors')
+
+test.type(setScriptDefault, 'function', 'command is function')
+test.equal(setScriptDefault.completion, require('../../lib/utils/completion/none.js'), 'empty completion')
+test.equal(setScriptDefault.usage, 'npm set-script [<script>] [<command>]', 'usage matches')
+test.test('fails on invalid arguments', (t) => {
+  const setScript = requireInject('../../lib/set-script.js', {
+    fs: {},
+    npmlog: {},
+  })
+  t.plan(3)
+  setScript(['arg1'], (fail) => t.match(fail, /Expected 2 arguments: got 1/))
+  setScript(['arg1', 'arg2', 'arg3'], (fail) => t.match(fail, /Expected 2 arguments: got 3/))
+  setScript(['arg1', 'arg2', 'arg3', 'arg4'], (fail) => t.match(fail, /Expected 2 arguments: got 4/))
+})
+test.test('fails if run in postinstall script', (t) => {
+  var originalVar = process.env.npm_lifecycle_event
+  process.env.npm_lifecycle_event = 'postinstall'
+  const setScript = requireInject('../../lib/set-script.js', {
+    fs: {},
+    npmlog: {},
+  })
+  t.plan(1)
+  setScript(['arg1', 'arg2'], (fail) => t.equal(fail.toString(), 'Error: Scripts can’t set from the postinstall script'))
+  process.env.npm_lifecycle_event = originalVar
+})
+test.test('fails when package.json not found', (t) => {
+  const setScript = requireInject('../../lib/set-script.js', {
+    '../../lib/npm.js': {
+      localPrefix: 'IDONTEXIST',
+    },
+  })
+  t.plan(1)
+  setScript(['arg1', 'arg2'], (fail) => t.equal(fail.toString(), 'package.json not found'))
```

Comment:
ooops, I forgot this one:

```suggestion
  setScript(['arg1', 'arg2'], (fail) => t.match(fail, /package.json not found/))
```